### PR TITLE
NavbarButtonExpansion

### PR DESCRIPTION
**Pull request by itsliterallymonique**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### Description
changed navbar to be able to have more than 4 categories. It will go to the next / previous four categories when you click the previous and back buttons. It will also disappear if you reached the end or start of the categories.

[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # "This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation."
🛍 CC: @EthanAuyeung

### CODE REVIEW

File: src/api/supabase/queries/button_queries.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    .from('storefront_buttons')
+    .select('*');
+  if (error) {
+    throw new Error(`Error fetching buttons: ${error.message}`);
+  }
+
+  return buttons;
+}
+
+export async function fetchButton() {
```
Comment (BuyankhuuTsCAl) [inline src/api/supabase/queries/button_queries.ts:15]: what is this function for?

File: src/app/storefront/StoreFrontNavBar.tsx [hunk 2]
```diff
@@ -1,7 +1,9 @@
+/* eslint-disable react/button-has-type */
 import Image from 'next/image';
 import Link from 'next/link';
 import React, { useEffect, useState } from 'react';
 
+import { fetchButoonCategories } from '@/api/supabase/queries/button_queries';
```
Comment (BuyankhuuTsCAl) [inline src/app/storefront/StoreFrontNavBar.tsx]: make sure you are spelling things right?

File: src/app/storefront/StoreFrontNavBar.tsx [hunk 3]
```diff
@@ -12,29 +14,42 @@ import {
...
+import { Product, StorefrontButtons } from '../../schema/schema';
 
-import { ButtonsContainer } from './styles';
-
-import { buttons } from './buttonValues';
+import {
+  ButtonsContainer,
+  FrontButton,
+  BackButton,
+  Front,
```
Comment (BuyankhuuTsCAl) [inline src/app/storefront/StoreFrontNavBar.tsx]: this can be confusing since there's frontbutton and front

File: src/app/storefront/StoreFrontNavBar.tsx [hunk 4]
```diff
@@ -62,24 +125,30 @@ export default function StoreFrontNavBar(props: {
...
+        <BackButton onClick={handleNext} $reachedEnd={reachedEnd}>
+          <Back />
+        </BackButton>
       </ButtonsContainer>
 
       <ButtonsDiv>
         <Link href="../profileScreen">
           <UserProfileIcon />
-          <p>User</p>
+          <p>Users</p>
```
Comment (BuyankhuuTsCAl) [inline src/app/storefront/StoreFrontNavBar.tsx]: make sure that text is being surrounded by text components

**Review by BuyankhuuTsCAl — COMMENTED** [review]
Nice job! Please address the small commnets so i can merge your code in!

**Comment by BuyankhuuTsCAl** [comment]
question: now that you added a table for the buttons, do you think we need the buttonValues column anymore?
